src/JsonLd/ContextBuilder.php

```diff
@@ -87,9 +87,45 @@ public function getEntrypointContext(int $referenceType = UrlGeneratorInterface:
      */
     public function getResourceContext(string $resourceClass, int $referenceType = UrlGeneratorInterface::ABS_PATH): array
     {
-        $context = $this->getBaseContext($referenceType);
+        $metadata = $this->resourceMetadataFactory->create($resourceClass);
+        if (null === $shortName = $metadata->getShortName()) {
+            return [];
+        }
+
+        return $this->getResourceContextWithShortname($resourceClass, $referenceType, $shortName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getResourceContextUri(string $resourceClass, int $referenceType = UrlGeneratorInterface::ABS_PATH): string
+    {
         $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
-        $shortName = $resourceMetadata->getShortName();
+
+        return $this->urlGenerator->generate('api_jsonld_context', ['shortName' => $resourceMetadata->getShortName()], $referenceType);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAnonymousResourceContext($object, array $context, int $referenceType = UrlGeneratorInterface::ABS_PATH): array
+    {
+        $id = $context['iri'] ?? '_:'.spl_object_id($object);
```

Comment:
Using `spl_object_id()` drops support for PHP 7.1. Otherwise you have to add `symfony/polyfill-php72` as dependency...